src/test/mimc_mp_test.cpp

```diff
@@ -11,6 +11,141 @@ typedef libff::Fr<ppT> FieldT;
 namespace
 {
 
+// Testing that (15212  + 98645 + 216319)**7 =
+// 427778066313557225181231220812180094976
+TEST(TestRound, TestTrueNoAddKToResult)
+{
+    libsnark::protoboard<FieldT> pb;
+    libsnark::pb_variable<FieldT> in_x;
+    libsnark::pb_variable<FieldT> in_k;
+
+    FieldT in_C = FieldT("216319");
+    in_x.allocate(pb, "x");
+    in_k.allocate(pb, "k");
+    pb.val(in_x) = FieldT("15212");
+    pb.val(in_k) = FieldT("98645");
+
+    MiMCe7_round_gadget<FieldT> round_gadget(
+        pb, in_x, in_k, in_C, false, "round_gadget");
+    round_gadget.generate_r1cs_witness();
+    round_gadget.generate_r1cs_constraints();
+
+    FieldT expected_out = FieldT("427778066313557225181231220812180094976");
+    ASSERT_TRUE(expected_out == pb.val(round_gadget.result()));
+}
+
+// Testing that (15212  + 98645 + 216319)**7 + 98645 =
+// 427778066313557225181231220812180193621
+TEST(TestRound, TestTrueAddKToResult)
+{
+    libsnark::protoboard<FieldT> pb;
+    libsnark::pb_variable<FieldT> in_x;
+    libsnark::pb_variable<FieldT> in_k;
+
+    FieldT in_C = FieldT("216319");
+    in_x.allocate(pb, "x");
+    in_k.allocate(pb, "k");
+    pb.val(in_x) = FieldT("15212");
+    pb.val(in_k) = FieldT("98645");
+
+    MiMCe7_round_gadget<FieldT> round_gadget(
+        pb, in_x, in_k, in_C, true, "round_gadget");
+    round_gadget.generate_r1cs_witness();
+    round_gadget.generate_r1cs_constraints();
+
+    FieldT expected_out = FieldT("427778066313557225181231220812180193621");
+    ASSERT_TRUE(expected_out == pb.val(round_gadget.result()));
+}
+
+TEST(TestRound, TestFalseNoAddKToResult)
```

Comment:
Isn't a dummy test case? If the one above passes, then this one will fail for sure since `427778066313557225181231220812180193621` =/= `4277780663135572251`
(Likewise for the next test)

Reply:
Changed the"TestFalseX" tests to check the circuit output are not constant.